### PR TITLE
RELATED: RAIL-4518 fix drills on empty values

### PR DESCRIPTION
**Pull request by no23reason**
Removes the fixEmptyHeaderItems usage. This means that drilling on empty item returns item with name correctly set to "" instead of localized "(empty value)". This in turn makes drill to insight work properly in the Dashboard component with empty values.

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by no23reason** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9906/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11829/ : SUCCESS in 23m 02s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1833/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9907/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1833/ : UNSTABLE in 34m 21s

**Comment by no23reason** [comment]
Failing screenshots have nothing to do with my changes.

File: libs/sdk-ui-charts/src/highcharts/chartTypes/_chartOptions/chartOptionsBuilder.ts [hunk 1]
```diff
@@ -324,6 +331,7 @@ export function getChartOptions(
     dataView: IDataView,
     chartConfig: IChartConfig,
     drillableItems: IHeaderPredicate[],
+    emptyHeaderName: string,
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-charts/src/highcharts/chartTypes/_chartOptions/chartOptionsBuilder.ts:334]: should not be named as emptyElement/Value? It represents single element, not whole attributeHeader, in not it?

File: libs/sdk-ui-pivot/src/impl/data/rowFactory.ts [hunk 2]
```diff
@@ -53,7 +53,9 @@ function getCell(
     if (isResultAttributeHeader(rowHeaderDataItem)) {
         return {
             ...cell,
-            value: rowHeaderDataItem.attributeHeaderItem.name,
+            value:
+                rowHeaderDataItem.attributeHeaderItem.name ||
+                `(${intl.formatMessage({ id: "visualization.emptyValue" })})`, // TODO RAIL-4360 distinguish between empty and null
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-pivot/src/impl/data/rowFactory.ts:58]: should not we provide string as external param in table too?

File: libs/sdk-ui/src/base/vis/drilling.ts [hunk 3]
```diff
@@ -145,7 +145,7 @@ function convertToEmpty(attributeItem: IResultAttributeHeader) {
         attributeHeaderItem: {
             ...attributeHeaderItem,
             // Send empty string for not, need to be updated for NULL in future
-            name: isEmpty ? "" : attributeHeaderItem.name,
+            name: isEmpty ? "" : attributeHeaderItem.name ?? "",
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui/src/base/vis/drilling.ts:148]: please add brackets to improve readability

**Review by ivan-nejezchleb — CHANGES_REQUESTED** [review]

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9909/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11831/ : FAILURE in 20m 16s

File: libs/sdk-ui/src/base/vis/drilling.ts (hunk 3 above)
Comment (no23reason) [reply libs/sdk-ui/src/base/vis/drilling.ts:148]: Unfortunately Prettier strips those.

File: libs/sdk-ui-charts/src/highcharts/chartTypes/_chartOptions/chartOptionsBuilder.ts (hunk 1 above)
Comment (no23reason) [reply libs/sdk-ui-charts/src/highcharts/chartTypes/_chartOptions/chartOptionsBuilder.ts]: In fact it is a header title related to the given value so it is a header. I changed the naming slightly and updated the TSDocs.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9913/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11836/ : SUCCESS in 20m 47s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9915/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11838/ : SUCCESS in 17m 43s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9916/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11840/ : FAILURE in 4m 13s

File: libs/sdk-ui-pivot/src/impl/data/rowFactory.ts (hunk 2 above)
Comment (no23reason) [reply libs/sdk-ui-pivot/src/impl/data/rowFactory.ts]: unified into one utility function

**Review by ivan-nejezchleb — APPROVED** [review]

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3022

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2834/ : SUCCESS in 1m 05s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3139/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1979/ : SUCCESS in 16m 20s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11842/ : FAILURE in 11m 15s

**Review by ivan-nejezchleb — APPROVED** [review]

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3022

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Pull request merged**